### ISSUE TITLE
预测阶段不能将结果也在网络中反向传播吧，会过拟合的

### ISSUE THREAD

**Issue by mJackie** [gaussic/text-classification-cnn-rnn#117]
test方法传入的数据不能用作更新网络参数，这点代码里没有做到

**Comment by gaussic** [comment]
给跪了，test方法并没有任何的反向传播过程，更不要说更新网络参数了。

**Comment by mJackie** [comment]
代码中test阶段的 feed_dic 如下
```
        feed_dict = {
            model.input_x: x_test[start_id:end_id],
            model.keep_prob: 1.0
        }
```
model.keep_prob 应该改为0 吧， 这样drop_out掉以后就不会反向传播了

**Comment by gaussic** [comment]
https://stackoverflow.com/questions/44971349/how-to-turn-off-dropout-for-testing-in-tensorflow

请自行测试把 keep_prob 设置为 0 的结果。

另外这里的 dropout 的计算与反向传播也没有任何关系。

**Comment by mJackie** [comment]
又差了些资料，应该这样理解：
train 的的时候 session.run(model.optim, feed_dict=feed_dict) 获取了model.optim，所以会去优化，反向传播
test阶段只获取了accuracy，没有去获取model.optim这个tensor
